### PR TITLE
Support PHPStan 2.0

### PR DESCRIPTION
**Pull request by zonuexe**
All packages that are not compatible with PHPStan 2.0 have been temporarily removed.

refs https://github.com/rectorphp/rector/issues/8815

### CODE REVIEW

**Comment by zonuexe** [comment]
I thought I needed a `RuleErrorBuilderRector` that works with PHPStan 1.x 😜

**Comment by zonuexe** [comment]
All tests pass in my environment.

スクリーンショット 2024-11-13 19 52 21

File: src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php [hunk 1]
```diff
@@ -32,18 +34,11 @@ public function __construct(
...
     public function getNodeType(): string
     {
         return FileNode::class;
     }
 
-    /**
-     * @param FileNode $node
-     * @return string[]
-     */
     public function processNode(Node $node, Scope $scope): array
```
Comment (TomasVotruba) [inline src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php:42]: The `@param` should remain, so IDE can fill the mtehods on `$node`

Comment (TomasVotruba) [reply src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php]: My lattest PHPStorm still doesn't. What IDE do you use?

Screenshot from 2024-11-13 12-16-33

**Comment by zonuexe** [comment]
I have to hurry up and catch a long-distance bus now, but I've finished what I can do.

File: src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php [hunk 2]
```diff
@@ -11,11 +11,13 @@
 use PHPStan\Analyser\Scope;
 use PHPStan\Node\FileNode;
 use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
 use Symplify\RuleDocGenerator\Contract\DocumentedRuleInterface;
 use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
 use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
 
 /**
+ * @implements <FileNode>
```
Comment (ondrejmirtes) [inline src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php:20]: The autocompletion does not work because this should say `@implements Rule<FileNode>`.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php (hunk 1 above)
Comment (ondrejmirtes) [reply src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php]: This is why: https://github.com/symplify/phpstan-rules/pull/144#pullrequestreview-2435518909

File: src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php (hunk 2 above)
Comment (zonuexe) [reply src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php]: Thanks!

File: src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php [hunk 3]
```diff
@@ -17,7 +17,7 @@
 use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
 
 /**
- * @implements <FileNode>
+ * @implements Node<FileNode>
```
Comment (JoyceBabu) [inline src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php:20]: Shouldn't this be `Rule<FileNode>`

Comment (ondrejmirtes) [reply src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php:20]: Yes. Running PHPStan on the code will report that.

File: composer.json [hunk 4]
```diff
@@ -7,21 +7,16 @@
...
+        "phpstan/phpstan": "^2.0",
         "symplify/rule-doc-generator-contracts": "^11.2"
     },
     "require-dev": {
         "nikic/php-parser": "^4.19",
-        "symplify/phpstan-extensions": "^11.4",
         "symplify/rule-doc-generator": "^12.2",
         "phpunit/phpunit": "^10.5",
         "symfony/framework-bundle": "6.1.*",
-        "rector/rector": "^1.2.3",
```
Comment (samsonasik) [inline composer.json:19]: could you use 

```php
        "rector/rector": "dev-main as 1.2.10",
```

to test rector for CI?

Comment (zonuexe) [reply composer.json:19]: I applied it to https://github.com/symplify/phpstan-rules/pull/144/commits/f83aafa22ee855df2305aab9be427dec162c278e

File: src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php (hunk 3 above)
Comment (zonuexe) [reply src/Rules/ForbiddenMultipleClassLikeInOneFileRule.php]: Fixed

File: src/Rules/CheckClassNamespaceFollowPsr4Rule.php [hunk 5]
```diff
@@ -87,7 +88,7 @@ public function processNode(Node $node, Scope $scope): array
         $namespacePart = substr($namespaceBeforeClass, 0, -1);
         $errorMessage = sprintf(self::ERROR_MESSAGE, $namespacePart);
 
-        return [$errorMessage];
+        return [RuleErrorBuilder::message($errorMessage)->build()];
```
Comment (samsonasik) [inline src/Rules/CheckClassNamespaceFollowPsr4Rule.php:91]: identifier name should be set?

```
->identifier('check.classnamespacepsr4')
```

File: .github/workflows/code_analysis.yaml [hunk 6]
```diff
@@ -22,6 +22,7 @@ jobs:
 
                     -
                         name: 'Rector'
+                        if: ${{ false }} # Temporarily disabled as it is not compatible with PHPStan 2.0
```
[the remark this answers is not in the document]
Comment (zonuexe) [reply .github/workflows/code_analysis.yaml]: reverted https://github.com/symplify/phpstan-rules/pull/144/commits/a6be6a10f77e214f2236593d9f739579676e8975 https://github.com/symplify/phpstan-rules/pull/144/commits/d8fb908ab8ff364b3b69577fbaed610c65e30199

File: src/Rules/CheckClassNamespaceFollowPsr4Rule.php (hunk 5 above)
Comment (zonuexe) [reply src/Rules/CheckClassNamespaceFollowPsr4Rule.php]: fixed https://github.com/symplify/phpstan-rules/pull/144/commits/c8b251300eed3a078648f2a8220e841a7516ac8e

File: src/Rules/AnnotateRegexClassConstWithRegexLinkRule.php [hunk 7]
```diff
@@ -69,7 +67,7 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
-        return [self::ERROR_MESSAGE];
+        return [RuleErrorBuilder::message(self::ERROR_MESSAGE)->build()];
```
Comment (samsonasik) [inline src/Rules/AnnotateRegexClassConstWithRegexLinkRule.php:70]: add identifier here as well

File: src/Rules/CheckRequiredInterfaceInContractNamespaceRule.php [hunk 8]
```diff
@@ -53,7 +50,7 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
-        return [self::ERROR_MESSAGE];
+        return [RuleErrorBuilder::message(self::ERROR_MESSAGE)->build()];
```
Comment (samsonasik) [inline src/Rules/CheckRequiredInterfaceInContractNamespaceRule.php:53]: add identifier here as well

File: src/Rules/ClassNameRespectsParentSuffixRule.php [hunk 9]
```diff
@@ -132,7 +131,7 @@ private function processClassNameAndShort(ClassReflection $classReflection): arr
             }
 
             $errorMessage = sprintf(self::ERROR_MESSAGE, $expectedSuffix);
-            return [$errorMessage];
+            return [RuleErrorBuilder::message($errorMessage)->build()];
```
Comment (samsonasik) [inline src/Rules/ClassNameRespectsParentSuffixRule.php:134]: add identifier here as well

File: src/Rules/Complexity/ForbiddenArrayMethodCallRule.php [hunk 10]
```diff
@@ -58,7 +56,7 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
-        return [self::ERROR_MESSAGE];
+        return [RuleErrorBuilder::message(self::ERROR_MESSAGE)->build()];
```
Comment (samsonasik) [inline src/Rules/Complexity/ForbiddenArrayMethodCallRule.php:59]: add identifier here as well

File: src/Rules/Domain/RequireAttributeNamespaceRule.php [hunk 11]
```diff
@@ -74,6 +75,6 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
-        return [self::ERROR_MESSAGE];
+        return [RuleErrorBuilder::message(self::ERROR_MESSAGE)->build()];
```
Comment (samsonasik) [inline src/Rules/Domain/RequireAttributeNamespaceRule.php:78]: add identifier here as well

**Comment by mitelg** [comment]
hey @zonuexe 

thanks for your current work! :muscle: we are currently waiting for this extension to be compatible with PHPStan 2, so I want to offer my help :slightly_smiling_face: let me know, if I can somehow support here

**Comment by samsonasik** [comment]
I cherry-picked your commits at new PR to continue work on it:

- https://github.com/symplify/phpstan-rules/pull/145

**Comment by staabm** [comment]
this PR can be closed I guess?

**Comment by samsonasik** [comment]
Yes 👍

**Pull request closed without merging**